students/rob_sanchez/Lesson 3/mailroom.py

```diff
@@ -0,0 +1,166 @@
+        donor_dict[donor_name]["donations"][0]["amount"] = float(amt_input)
+        send_email(donor_name, amt_input)
+    else:
+        amt_input = input("\nDonation amount:\n")
+        history = donor_dict[donor_name]["donations"]
+        num_of_donations = []
+
+        for item in history:
+            num_of_donations.append(item.get('amount'))
+
+        new_index = len(num_of_donations)
+
+        donor_dict[donor_name]["donations"].append(
+                            {"donation-number": '', "amount": ''})
+        donor_dict[donor_name]["donations"][new_index]["donation-number"] = new_index + 1
+        donor_dict[donor_name]["donations"][new_index]["amount"] = float(amt_input)
+
+        send_email(donor_name, amt_input)
+
+
+# Creates a summary report of the donations
+def create_report():
+    headers = ["Donor Name", "Total Given", "Num Gifts", "Average Gift"]
+    print(("{:<30} | {:^15} | {:^15} | {:^15}").format(*headers))
+    print("-"*82)
+
+    donor_report = get_summary()
+
+    for item in donor_report:
+        print(("{:<30} ${:>17} {:>16} ${:>14}").format(*item))
+
+    welcome()
+
+
+# Helper methods
+def welcome_options():
+    # Menu options
+    options = "1. Send a Thank You\n2. Create a Report\n3. Quit\n"
+    selection_options = ('1', '2', '3')
+
+    # Selection value
+    selection_no = input(options)
+
+    # Keep prompting user until a matching value is found
+    while selection_no not in selection_options:
+        print("\nInvalid Input! Try again.\n")
+        selection_no = input(options)
+    return selection_no
+
+
+# Returns a list of formatted values
+def print_list(in_val):
+    form_string = ", ".join(["{:s}"] * len(in_val))
+    return form_string.format(*in_val)
+
+
+# Sends an email to the specified donor
+def send_email(donor, amount):
+    greeting = "\nDear {},\n\n".format(donor)
+    body = ("I would like to personally thank you for your generours donation"
+            " of ${:.2f} to our charity organization.\nYour support allows us "
+            "to continue supporting more individuals in need of our services."
+            "\n\nSincerely,\nCharity Inc.\n").format(float(amount))
+    print(greeting, body)
+    welcome()
+
+
+# Returns donation summary of each donor
+def get_summary():
+    donor_names = []
+    total_given = []
+    num_gifts = []
+    i = 0
+
+    for donor, value in donor_dict.items():
+        history = donor_dict[donor]["donations"]
+        donor_names.append([])
+        donor_names[i].append(donor)
+        for item in history:
+            total_given.append(float(item.get('amount')))
+            num_gifts.append(item.get('donation-number'))
+        donor_names[i].append("{:.2f}".format(sum_values(total_given)))
+        donor_names[i].append(len(num_gifts))
+        donor_names[i].append("{:.2f}".format(sum_values(total_given)/len(num_gifts)))
+        total_given = []
+        num_gifts = []
+        i += 1
+    return donor_names
```

Comment:
the report summary should be sorted (check instructions); I'm not seeing the sorting.